.github/workflows/release.main.kts

```diff
@@ -0,0 +1,134 @@
+import io.github.typesafegithub.workflows.actions.nexusactions.CreateNexusStagingRepo
+import io.github.typesafegithub.workflows.actions.nexusactions.DropNexusStagingRepoV1
+import io.github.typesafegithub.workflows.actions.nexusactions.ReleaseNexusStagingRepo
+import io.github.typesafegithub.workflows.domain.AbstractResult
+import io.github.typesafegithub.workflows.domain.JobOutputs
+import io.github.typesafegithub.workflows.domain.RunnerType
+import io.github.typesafegithub.workflows.domain.triggers.WorkflowDispatch
+import io.github.typesafegithub.workflows.dsl.expressions.Contexts
+import io.github.typesafegithub.workflows.dsl.expressions.expr
+import io.github.typesafegithub.workflows.dsl.workflow
+
+val SONATYPE_USERNAME by Contexts.secrets
+val SONATYPE_PASSWORD by Contexts.secrets
+val SONATYPE_STAGING_PROFILE_ID by Contexts.secrets
+val SIGNING_KEY_ID by Contexts.secrets
+val SIGNING_KEY by Contexts.secrets
+val SIGNING_PASSWORD by Contexts.secrets
+
+workflow(
+    name = "Publish release to Maven Central",
+    on = listOf(
+        WorkflowDispatch()
+    ),
+    sourceFile = __FILE__,
+) {
+    val stagingRepoJob =
+        job(
+            id = "create-staging-repo",
+            name = "Create staging repository",
+            runsOn = RunnerType.UbuntuLatest,
+            outputs = object : JobOutputs() {
+                var repositoryId: String by output()
+            }
+        ) {
+            val createRepo = uses(
+                action = CreateNexusStagingRepo(
+                    username = expr { SONATYPE_USERNAME },
+                    password = expr { SONATYPE_PASSWORD },
+                    stagingProfileId = expr { SONATYPE_STAGING_PROFILE_ID },
+                )
+            )
+            jobOutputs.repositoryId = createRepo.outputs.repositoryId
+        }
+    val publishJob =
+        job(
+            id = "publish-artifacts",
+            runsOn = RunnerType.MacOSLatest,
+            needs = listOf(stagingRepoJob),
+        ) {
+            uses(action = Checkout())
+            uses(
+                name = "Set up JDK",
+                action = SetupJava(
+                    javaVersion = "11",
+                    distribution = SetupJava.Distribution.Zulu,
+                    cache = SetupJava.BuildPlatform.Gradle,
+                ),
+            )
+            uses(
+                name = "Cache Kotlin Konan",
+                action = Cache(
+                    path = listOf(
+                        "~/.konan/**/*",
+                    ),
+                    key = "kotlin-konan-${expr { runner.os }}",
+                ),
+            )
+            uses(
+                name = "Set up Gradle",
+                action = ActionsSetupGradle(
+                    gradleVersion = "wrapper",
+                ),
+            )
+            run(
+                name = "Publish",
+                command = "./gradlew publishAllPublicationsToSonatypeReleaseRepository --stacktrace",
```

Comment:
question: What do you think about adding the `--info` option here as well? It should help to investigate what was published and where (in case of issues with the publication)

Reply:
`--info` tends to be a little too noisy for everyday use.

When it comes to publishing, usually the Gradle tasks are stable and run correctly. It's usually configuring the correct properties in build scripts and Sonatype that are troublesome, and _hopefully_ `--stacktrace` provides enough help for that.